lib/model.js

```diff
@@ -1354,15 +1356,37 @@ Model.init = function init(callback) {
   const Promise = PromiseProvider.get();
   const autoIndex = utils.getOption('autoIndex',
     this.schema.options, this.db.config, this.db.base.options);
-  const autoCreate = utils.getOption('autoCreate',
-    this.schema.options, this.db.config, this.db.base.options);
 
   const _ensureIndexes = autoIndex ?
     cb => this.ensureIndexes({ _automatic: true }, cb) :
     cb => cb();
-  const _createCollection = autoCreate ?
-    cb => this.createCollection({}, cb) :
-    cb => cb();
+
+  const model = this;
+  const _wrapCollHelper = function(fn) {
+    return function(cb) {
+      const conn = model.db;
+
+      return promiseOrCallback(cb, cb => {
+        immediate(() => {
+          if ((conn.readyState === STATES.connecting || conn.readyState === STATES.disconnected) && conn._shouldBufferCommands()) {
+            conn._queue.push({ fn: fn, ctx: conn, args: [cb] });
+          } else {
+            try {
+              fn.apply(conn, [cb]);
+            } catch (err) {
+              return cb(err);
+            }
+          }
+        });
+      });
+    };
+  };
+  const _createCollection = _wrapCollHelper(function _createCollection(cb) {
+    const autoCreate = utils.getOption('autoCreate',
```

Comment:
Also, let's not add a separate `_wrapCollHelper()` function, since that function won't be reused anywhere. Just add the `if ((conn.readyState === STATES.connecting || conn.readyState === STATES.disconnected) && conn._shouldBufferCommands())` check directly into `_createCollection()`.